### ISSUE TITLE
chore: use yamux instead of mplex 

### ISSUE THREAD

**Issue by weboko** [waku-org/js-waku#1771]
This is a **change request**

## Proposed Solutions

Use `yamux` here:
https://github.com/waku-org/js-waku/blob/c81872ae0e27331fc2c9249f310c3901cf06d5ce/packages/sdk/src/create.ts#L215

## Notes

From the thread - https://github.com/waku-org/go-waku/pull/851#discussion_r1387565204

**Comment by danisharora099** [comment]
Blocked until nwaku releases support for yamux: https://github.com/waku-org/nwaku/pull/2518

**Comment by weboko** [comment]
@danisharora099 seems to be unblocked by https://github.com/waku-org/nwaku/releases/tag/v0.27.0

**Comment by danisharora099** [comment]
Yes, the PR upgraded nwaku to 0.27 but dials seem to be failing. Will come back to this and investigate

**Comment by danisharora099** [comment]
Blocked by https://github.com/libp2p/js-libp2p/issues/2505 / https://github.com/ChainSafe/js-libp2p-yamux/issues/91

**Comment by weboko** [comment]
Moving to `in progress` to check it out once more.

**Comment by weboko** [comment]
![Image](https://github.com/user-attachments/assets/82f31681-e4de-43c1-8c3f-2f311a99a4ed)

**Comment by weboko** [comment]
Seems to be coming from https://github.com/vacp2p/nim-libp2p/blob/ebbfb63c17bc023d77981777a8134d33305c9a68/libp2p/muxers/yamux/yamux.nim#L622
